src/utils.jl

```diff
@@ -532,6 +532,90 @@ julia> zeros_like(x, Float64)
 zeros_like(x::AbstractArray, T::Type, sz=size(x)) = fill!(similar(x, T, sz), 0)
 zeros_like(x::AbstractArray, sz=size(x)) = zeros_like(x, eltype(x), sz)
 
+"""
+    rand_like([rng=GLOBAL_RNG], x, [element_type=eltype(x)], [dims=size(x)])
+
+Create an array with the given element type and size, based upon the given source array `x`.
+All element of the new array will be set to a random value.
+The second and third arguments are both optional, defaulting to the given array's eltype and
+size. The dimensions may be specified as an integer or as a tuple argument.
+
+See also `Base.rand` and [`randn_like`](@ref).
+
+# Examples
+
+```julia-repl
+julia> x = rand(Float32, 2)
```

Comment:
```suggestion
julia> x = ones(Float32, 2)
```
and same for the cuda example